### PR TITLE
Moving Hapi Plugin to Instrumentation

### LINKED ISSUE #255: Convert hapi plugin to instrumentation
- Change implementation to use [InstrumentationBase](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts#L30)

### PR DESCRIPTION
**Pull request by obecny**
## Which problem is this PR solving?

- Fixes #255

## Short description of the changes

- Convert hapi plugin to instrumentation

### CODE REVIEW

File: examples/hapi/package.json [hunk 1]
```diff
@@ -35,8 +35,9 @@
     "@opentelemetry/api": "^0.18.0",
     "@opentelemetry/exporter-jaeger": "^0.18.0",
     "@opentelemetry/exporter-zipkin": "^0.18.0",
-    "@opentelemetry/hapi-instrumentation": "^0.13.1",
     "@opentelemetry/instrumentation": "^0.18.0",
+    "@opentelemetry/instrumentation-hapi": "^0.13.1",
+    "@opentelemetry/instrumentation-http": "^0.18.0",
     "@opentelemetry/node": "^0.18.0",
     "@opentelemetry/plugin-http": "^0.18.0",
```
Comment (Flarna) [inline examples/hapi/package.json:42]: I think this should be removed

File: plugins/node/opentelemetry-instrumentation-hapi/README.md [hunk 2]
```diff
@@ -25,38 +25,42 @@ To load a specific instrumentation (Hapi in this case), specify it in the regist
 
 ```js
 const { NodeTracerProvider } = require('@opentelemetry/node');
```
Comment (Flarna) [inline plugins/node/opentelemetry-instrumentation-hapi/README.md:27]: replace `hapi-instrumentation` by `instrumentation-hapi` a few lines above (github doesn't allow to comment there)

**Review by Flarna — APPROVED** [review]

**Review by vmarchaud — APPROVED** [review]

**Review by johnbley — APPROVED** [review]

**Comment by nmiddendorff** [comment]
I just attempted to run the hapi example locally and had trouble resolving `@opentelemetry/instrumentation-hapi`. Does this need to switch back to `@opentelemetry/hapi-instrumentation`?

**Comment by obecny** [comment]
`@opentelemetry/instrumentation-hapi` hasn't been released yet

**Comment by Flarna** [comment]
@nmiddendorff see https://github.com/open-telemetry/opentelemetry-js-contrib/pull/409

**Pull request merged**